src/algebra/group_ring_action.lean

```diff
@@ -130,6 +130,15 @@ end semiring
 
 section ring
 
+lemma smul_cancel_of_non_zero_divisor {M R : Type*} [monoid M] [ring R] [distrib_mul_action M R]
```

Comment:
Don't we have a typeclass for this?

Reply:
cc @Vierkantor. I think we should at least mention `no_zero_smul_divisors` in the docstring even if we don't use it here

Reply:
Also this is the wrong file, as this is only supposed to be about `mul_semiring_action`

Reply:
The top of the file says
```
There is no separate typeclass for group acting on rings, group acting on fields, etc.
They are all grouped under `mul_semiring_action`.
```
Do you mean this should be earlier in the import hierarchy, in a file about `distrib_mul_action`?